src/AspNetCoreModuleV2/CommonLib/GlobalVersionUtility.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "stdafx.h"
+#include <experimental/filesystem>
+
+namespace fs = std::experimental::filesystem;
+
+// throws runtime error if no request handler versions are installed.
+std::wstring
+GlobalVersionUtility::GetGlobalRequestHandlerPath(PCWSTR pwzAspNetCoreFolderPath, PCWSTR pwzHandlerVersion, PCWSTR pwzHandlerName)
+{
+    std::wstring folderVersion(pwzHandlerVersion);
+    fs::path aspNetCoreFolderPath(pwzAspNetCoreFolderPath);
+
+    if (folderVersion.empty())
+    {
+        folderVersion = FindHighestGlobalVersion(pwzAspNetCoreFolderPath);
+    }
+
+    aspNetCoreFolderPath = aspNetCoreFolderPath
+        .append(folderVersion)
+        .append(pwzHandlerName);
+    return aspNetCoreFolderPath;
+}
+
+// Throw filesystem_error if directory_iterator can't iterate over the directory
+std::vector<fx_ver_t>
+GlobalVersionUtility::GetRequestHandlerVersions(PCWSTR pwzAspNetCoreFolderPath)
+{
```

Comment:
if passed in pointer is NULL, I think you will get AV. please make sure it is not NULL or at least do debug assert